src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -187,5 +190,102 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
                 FileSystemAclExtensions.SetAccessControl(fileStream, fileSecurity);
             }
         }
+
+        [Fact]
+        public void DirectoryInfo_Create_NullDirectoryInfo()
+        {
+            DirectoryInfo info = null;
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<ArgumentNullException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NullDirectorySecurity()
+        {
+            DirectoryInfo info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>(() => info.Create(null));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_NotFound()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"W:\\I\\Do\\Not\\Exist");
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<DirectoryNotFoundException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_AccessDenied()
+        {
+            DirectoryInfo info = new DirectoryInfo(@"C:\\Windows\\System32\\deniedaccess");
+            DirectorySecurity security = new DirectorySecurity();
+            Assert.Throws<UnauthorizedAccessException>(() => info.Create(security));
+        }
+
+        [Fact]
+        public void DirectoryInfo_Create_DefaultDirectorySecurity()
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+
+        [Theory]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
```

Comment:
Maybe throw in a deny just for fun? Although again that's testing existing code.

Reply:
Sure, I can add a different combination. I will have to investigate what is the expected result for it.